ibm/data_source_ibm_cis_certificates.go

```diff
@@ -0,0 +1,162 @@
+package ibm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/IBM/go-sdk-core/v3/core"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+const (
+	cisCertificates                   = "certificates"
+	cisCertificatesCertificates       = "certificates"
+	cisCertificatesCertificatesID     = "id"
+	cisCertificatesCertificatesHosts  = "hosts"
+	cisCertificatesCertificatesStatus = "status"
+	cisCertificatesPrimaryCertificate = "primary_certificate"
+	cisCertificatesType               = "type"
+	cisCertificateTypeDedicated       = "dedicated"
+)
+
+func dataIBMCISCertificates() *schema.Resource {
+	return &schema.Resource{
+		Read: dataIBMCISCertificatesRead,
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS object id",
+				Required:    true,
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Description:      "Associated CIS domain",
+				Required:         true,
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			cisCertificates: {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Collection of certificates",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "certificate identifier",
+						},
+						cisCertificateOrderID: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "certificate id",
+						},
+						cisCertificatesType: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "certificate type",
+						},
+						cisCertificateOrderHosts: {
+							Type:        schema.TypeList,
+							Description: "Hosts which certificate need to be ordered",
+							Required:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+						cisCertificateOrderStatus: {
+							Type:        schema.TypeString,
+							Description: "certificate status",
+							Computed:    true,
+						},
+						cisCertificatesPrimaryCertificate: {
```

Comment:
Primary_certificate is int in the api docs
Any specific reason for making it to string

Reply:
this will list both dedicated and custom certificate. some certificate are having integer certificate id. some has string type. so converted to make it as string.